.github/workflows/ci.yml

```diff
@@ -1,7 +1,11 @@
 name: CI
 
 on:
-  [push, pull_request, workflow_call]
+  workflow_call:
+  pull_request:
+  push:
+    branches: [main]
```

Comment:
Does this trigger CI only for the pushes to 'main' branch only? Personally,I look at CI runs for any branch (without opening PR), however, this indeed may be a waste of resources

Reply:
Correct. If you add `workflow_dispatch` you can also press a button in "actions" to launch a build on any commit/branch/whatever.